newIDE/app/src/UI/Menu/ContextMenu.js

```diff
@@ -68,7 +71,9 @@ class ElectronContextMenu extends React.Component {
   }
 
   open = (x, y) => {
-    this.menuImplementation.buildFromTemplate(this.props.buildMenuTemplate());
+    this.menuImplementation.buildFromTemplate(
+      this.props.buildMenuTemplate(this.props.i18n)
+    );
     this.menuImplementation.showMenu({
```

Comment:
If you read and apply my previous comment, you're now like "ok sure I added `<I18n>` inside MaterialUIContextMenu, but now how do I do it for ElectronContextMenu??".
Ideally in the future, we would be able to use a React hook for I18n: `const i18n = use18n();`... but for now we have to use what is called the version with a "render prop": `<I18n>{({i18n}) => ...}<I18n>`, which is a bit more painful to use in this case.

The solution? Make a ElectronContextMenuWrapper component and a ElectronContextMenu component.
Basically:

- ElectronContextMenuWrapper is what you export. It is a component that just do:
```
const ElectronContextMenuWrapper = (props) => {
  return <I18n>{({i18n}) => <ElectronContextMenu {...props} i18n={i18n} >}<I18n>;
}
```

(in other words, you pass all the props to the child component, and you add i18n.
- you also need to expose the `open` function. As we're trying to use React hooks now, use `useImperativeHandle` to expose functions to the outside world, and use a ref to the component:

```
const ElectronContextMenuWrapper = React.forwardRef((props, ref) => {
    const electronContextMenu = React.useRef(null);
    React.useImperativeHandle(ref, () => ({
      open: (x, y) => {
        if (electronContextMenu.current) electronContextMenu.current.open(x,y);
      },
    }));

  return <I18n>{({i18n}) => <ElectronContextMenu {...this.props} i18n={i18n} ref={electronContextMenu}>}<I18n>;
}
```
- You can now `export default (electron ? ElectronContextMenuWrapper : MaterialUIContextMenu);` and if everything is right, your ContextMenu will work both in the web-app and the electron app, giving `i18n` as parameter of `buildMenuTemplate`, so all the other components don't need to import `<I18n>`, which is nice to keep the code shorter :) 

- You're lucky because you don't need to add Flow typing because I never added them to ContextMenu :p I'll do that later though, for such refactoring it's very good to see if you've forgot a prop somewhere. For now, I'll put my trust in you testing that it works everywhere ;)